src/payment/strategies/adyenv2/adyenv2-payment-strategy.spec.ts

```diff
@@ -242,318 +116,337 @@ describe('AdyenV2PaymentStrategy', () => {
-            expect(paymentActionCreator.submitPayment).toHaveBeenLastCalledWith({
-                methodId: 'scheme',
-                paymentData: {
-                    nonce: JSON.stringify({
-                        threeDS2Token: 'token',
-                        paymentData: 'paymentData',
+        describe('#execute', () => {
+            const identifyShopperError = getAdditionalActionError(ResultCode.IdentifyShopper);
+            const challengeShopperError = getAdditionalActionError(ResultCode.ChallengeShopper);
+            let additionalActionComponent: AdyenComponent;
+
+            beforeEach(() => {
+                let handleOnAdditionalDetails: (additionalActionState: AdyenAdditionalActionState) => {};
+
+                additionalActionComponent = {
+                    mount: jest.fn(() => {
+                        handleOnAdditionalDetails({
+                            data: {
+                                resultCode: ResultCode.ChallengeShopper,
+                                action: 'adyenAction',
+                            },
+                            isValid: true,
+                        });
+
+                        return;
                     }),
-                },
+                    unmount: jest.fn(),
+                };
+
+                jest.spyOn(adyenCheckout, 'createFromAction')
+                    .mockImplementation(jest.fn((_type, options) => {
+                        const { onAdditionalDetails } = options;
+                        handleOnAdditionalDetails = onAdditionalDetails;
+
+                        return additionalActionComponent;
+                    }));
             });
-            expect(adyenCheckout.create).toHaveBeenLastCalledWith(
-                AdyenComponentType.ThreeDS2DeviceFingerprint,
-                componentOptions
-            );
-            expect(adyenCheckout.create).toHaveBeenCalledTimes(3);
-        });
 
-        it('returns 3DS2 IdentifyShopper flow and Adyen response onError', async () => {
-            additionalActionComponent = {
-                mount: jest.fn(() => {
-                    componentOptions.onError({
-                        errorCode: 'errorCode',
-                        message: 'errorMessage',
-                    });
+            it('throws an error when payment is not present', async () => {
+                await strategy.initialize(options);
 
-                    return;
-                }),
-                unmount: jest.fn(),
-            };
+                try {
+                    await strategy.execute(getOrderRequestBodyWithoutPayment());
+                } catch (error) {
+                    expect(error).toBeInstanceOf(PaymentArgumentInvalidError);
+                }
 
-            jest.spyOn(adyenV2ScriptLoader, 'load').mockReturnValue(Promise.resolve(adyenCheckout));
+                expect(adyenCheckout.create).toHaveBeenCalledTimes(2);
```

Comment:
is not obvious as why you are expecting this to be called twice, mainly because the title of the test doesn't mention anything around it.